packages/block-editor/src/components/block-list/style.scss

```diff
@@ -51,26 +51,29 @@
 		// Compensate for side UI.
 		padding-left: 0;
 		padding-right: 0;
-
-		// Compensate for block padding.
-		margin-left: -$block-padding;
-		margin-right: -$block-padding;
 	}
 }
 
-.block-editor-block-list__layout .block-editor-block-list__block {
-	position: relative;
+.block-editor-block-list__block .block-editor-block-list__layout.has-spacing {
+	// Compensate for block padding.
```

Comment:
I definitely emphathise with the complex block paddings. But this PR seems to go a bit farther than necessary in making things work. Perhaps it's worth waiting for #17877 to land before rebasing this on top? 

Alternately, and I suspect this is due to the onerous extra UI applied to child blocks, you could snatch the code I wrote for social links, that should allow you to change the margins and paddings just for this one block, while awaiting better improvements than adding a new `has-spacing` property. Specifically it starts here: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/social-links/editor.scss#L14

Try the social links block and see how child blocks are well spaced out. If you need my help feel free to ping and I'll take a look.